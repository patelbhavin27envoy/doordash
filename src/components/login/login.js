import React from "react";
import { Redirect , withRouter } from 'react-router-dom'
import './login.scss'

class Login extends React.Component {

    constructor(props) {
        super(props);
        this.state = {
            userName: "",
            isAuthenticated: false,
            user: null,
            isSubmitting: false,
            errorMessage: ""
        };

        this.onSubmit = this.onSubmit.bind(this);
        this.login = this.login.bind(this);
        this.toggleIsSubmitting = this.toggleIsSubmitting.bind(this);
        this.handleInputChange = this.handleInputChange.bind(this);
    }

    onSubmit (e) {
        if (this.state.userName !== "") {
            e.preventDefault();
            this.login();
        }
    };
    login () {
        this.toggleIsSubmitting();
        this.setState({
            isAuthenticated: true
        });
    };
    toggleIsSubmitting () {
        this.setState(prevState => ({
            isSubmitting: !prevState.isSubmitting
        }));
    };
    handleInputChange (e ) {
        this.setState({
            userName: e.target.value
        });
    };

    render() {
        if (this.state.isAuthenticated) {
            return (
                <Redirect
                to={{
                    pathname: "/chatwindow",
                        state: { userName: this.state.userName }
                }}
            />
        );
        }

        const isSubmitting = this.state.isSubmitting;


        return (
            <div className="App">
                <h1>Chat App</h1>
                <p>Create an account through your Door dashboard.</p>
                <form className="form" onSubmit={this.onSubmit}>
                    <input onChange={this.handleInputChange} type="text" className="userName"  name="userName" />
                    <span className="error">{this.state.errorMessage}</span>
                    {isSubmitting ? (
                           <span> loading ... </span>
                    ) : (
                    <input className="btn"
                        type="submit"
                        disabled={this.state.userName === ""}
                        value="Join the Door chat"
                            />
                    )}
                </form>
            </div>
    );
    }
}
export default Login;
