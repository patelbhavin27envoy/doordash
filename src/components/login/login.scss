@import '../../index';

.App {
  text-align: center;
  display: flex;
  width: 100%;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  height: 50vh;

  p {
    font-size: 12px;
    width: 50%;
  }

  .App-logo {
    animation: App-logo-spin infinite 0.5s linear;
    height: 10vmin;
  }

  input {
    border: 1px #{$color-grayMedium} solid;
  }

  .form input[type="submit"].btn {
    background: #{$color-redBright};
  }

}
