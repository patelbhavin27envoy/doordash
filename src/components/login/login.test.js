import React from 'react';
import renderer from 'react-test-renderer';

import { shallow, mount } from "enzyme";
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
Enzyme.configure({ adapter: new Adapter() });

import { Login } from '../login'

describe("Login Component :: ", () => {

    it("should render without throwing an error", () => {
        expect(
            renderer.create(<Login />)
        ).not.toBe(null);
    });

    it("should render ", () => {
        const component = mount(<Login />);
        expect(component).toMatchSnapshot();
    });

    it("should change name. ", () => {

        const component = mount(<Login />);

        expect(component.find('div').length).toEqual(1);
        expect(component.find('h1').length).toEqual(1);
        expect(component.find('h1').text()).toContain('Chat App');

        expect(component.find('input').length).toEqual(2);
        expect(component.find('p').length).toEqual(1);
        expect(component.find('p').text()).toContain('Create an account through your Door dashboard.');

        component.find('.userName').simulate('change', {target: {name: 'userName', value: 'John'}});
        component.find('.btn').simulate('click');

    });


    it("should check login button click. ", () => {

        const component = shallow(<Login />);
        const instance = component.instance();
        instance.state.userName ="test";
        jest.spyOn(instance, 'login');
        instance.login();
        expect(instance.login).toHaveBeenCalled();

        expect(component.state('isSubmitting')).toBe(true);

        instance.onSubmit({
            preventDefault: () => {
            }
        });

        expect(component.state('isAuthenticated')).toBe(true);
    });

});

