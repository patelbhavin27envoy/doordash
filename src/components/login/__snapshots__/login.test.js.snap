// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Component ::  should render  1`] = `ReactWrapper {}`;
